src/backend/mailbox_manager/utils/dimail.py

```diff
@@ -563,3 +580,35 @@ def fetch_domain_expected_config(self, domain):
                 exc_info=False,
             )
             return []
+
+    def reset_password(self, mailbox):
+        """Send a request to reset mailbox password."""
+        try:
+            response = session.post(
+                f"{self.API_URL}/domains/{mailbox.domain.name}/mailboxes/{mailbox.local_part}/reset_password/",
+                headers={"Authorization": f"Basic {self.API_CREDENTIALS}"},
+                verify=True,
+                timeout=self.API_TIMEOUT,
+            )
+        except requests.exceptions.ConnectionError as error:
+            logger.exception(
+                "Connection error while trying to reach %s.",
+                self.API_URL,
+                exc_info=error,
+            )
+            return []
```

Comment:
Be careful here: this will not raise and you will send an email "password reset" while nothing has been done.

I think you should test it actually.

Reply:
Done ! I added a test to check it fails properly.